kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -2677,6 +2679,79 @@ object KyuubiConf {
       .stringConf
       .createOptional
 
+  val ENGINE_DEPLOY_MODE: ConfigEntry[String] =
+    buildConf("kyuubi.engine.deploy.mode")
+      .doc("Configures the engine deploy mode, The value can be 'local', 'yarn'. " +
+        "In local mode, the engine operates on the same node as the KyuubiServer. " +
+        "In YARN mode, the engine runs within the Application Master (AM) container of YARN. " +
```

Comment:
This YARN mode is very interesting. If I understand correctly, this mode should also support the JDBC engine in the future.

Reply:
yea